src/vs/base/browser/ui/iconLabel/iconLabelHover.ts

```diff
@@ -127,27 +165,36 @@ export function setupCustomHover(hoverDelegate: IHoverDelegate, htmlElement: HTM
 		});
 	};
 	const mouseOverDomEmitter = dom.addDisposableListener(htmlElement, dom.EventType.MOUSE_OVER, () => showHoverDelayed(hoverDelegate.delay), true);
-	return {
+	const hover: ICustomHover = {
 		show: () => {
 			showHoverDelayed(0); // show hover immediately
 		},
 		hide: () => {
 			hideHover(true, true);
 		},
+		update: async newTooltip => {
+			markdownTooltip = newTooltip;
+			if (hoverWidget && !hoverWidget.isDisposed) {
+				let resolvedTooltip = await computeTooltip(markdownTooltip, CancellationToken.None);
```

Comment:
Given this is now a long running operation, I think we should wire in a `CancellationTokenSource` in a way that every subsequent call to `ICustomHover.update` cancels the previous run to prevent weird overlaps?

Reply:
Yes, makes sense.